src/raml.coffee

```diff
@@ -181,7 +181,8 @@ class @RamlParser
         node.value.forEach(([kind, item]) ->
           if item instanceof nodes.FileNode
             # schema inline inclusions
-            schemaNodes.push(item)
+            if kind.value == 'schema' && definedSchemaNames.indexOf(item.value) == -1
```

Comment:
the comment `# schema inline inclusions` belongs to `schemaNodes.push(item)`, please move it. also you could use one if block again by modifying the condition